crates/webidl/src/idl_type.rs

```diff
@@ -466,7 +474,16 @@ impl<'a> IdlType<'a> {
             IdlType::Uint32Array => Some(array("u32", pos)),
             IdlType::Float32Array => Some(array("f32", pos)),
             IdlType::Float64Array => Some(array("f64", pos)),
-
+            IdlType::ArrayBufferView | IdlType::BufferSource => {
+                if pos == TypePosition::Argument {
+                    // ArrayBuffer and BufferSource can be represented as `Object` or &[u8]
+                    IdlType::Union(vec![IdlType::Object, IdlType::Uint8Array]).to_syn_type(pos)
```

Comment:
I think for a simplification this can probably be `Object` as well actually in the sense that the "flatten" expansion happens below and here it's for things like attributes and whatnot (which largely only work with `Object`)

Reply:
done.